rpcperms/interceptor.go

```diff
@@ -103,19 +120,41 @@ func NewInterceptorChain(log btclog.Logger, noMacaroons,
 
 	return &InterceptorChain{
 		state:         startState,
+		ntfnServer:    subscribe.NewServer(),
 		noMacaroons:   noMacaroons,
 		permissionMap: make(map[string][]bakery.Op),
 		rpcsLog:       log,
+		quit:          make(chan struct{}),
 	}
 }
 
+func (r *InterceptorChain) Start() error {
+	var err error
+	r.started.Do(func() {
+		err = r.ntfnServer.Start()
+	})
+
+	return err
+}
+
+func (r *InterceptorChain) Stop() error {
+	var err error
+	r.stopped.Do(func() {
+		close(r.quit)
+		err = r.ntfnServer.Stop()
+	})
+
+	return err
+}
+
 // SetWalletUnlocked moves the RPC state from either walletNotCreated or
 // walletLocked to walletUnlocked.
 func (r *InterceptorChain) SetWalletUnlocked() {
 	r.Lock()
 	defer r.Unlock()
 
 	r.state = walletUnlocked
+	_ = r.ntfnServer.SendUpdate(r.state)
```

Comment:
nit: maybe worth adding a comment that we don't care about the error as it is only useful to indicate shutdown.

Reply:
I think it's okay without it for now, as this is a common pattern we use.